components/org.wso2.identity.event.common.publisher/src/main/java/org/wso2/identity/event/common/publisher/EventPublisher.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.common.publisher;
+
+import org.wso2.identity.event.common.publisher.model.EventContext;
+import org.wso2.identity.event.common.publisher.model.SecurityEventTokenPayload;
+
+/**
+ * The EventPublisher interface.
+ */
+public interface EventPublisher {
+
+    /**
+     * Publish a given event to the intermediate hub.
+     *
+     * @param payload      Event payload.
+     * @param eventContext  Event Context.
+     * @param eventUri     URI of the event.
+     * @throws Exception
+     */
+    void publish(SecurityEventTokenPayload payload, EventContext eventContext, String eventUri) throws Exception;
```

Comment:
event context has event uri, do we want to send it as a param ?

Also, it would be good to have a builder for event context

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/eb4cae044fe65111cb72917fd78af0619d056a02